app/src/main/java/org/onionshare/android/ui/OnionNotificationManager.kt

```diff
@@ -12,59 +13,95 @@ import android.os.Build.VERSION.SDK_INT
 import androidx.annotation.RequiresApi
 import androidx.core.app.NotificationCompat
 import androidx.core.app.NotificationCompat.CATEGORY_SERVICE
+import androidx.core.app.NotificationCompat.CATEGORY_STATUS
+import androidx.core.app.NotificationCompat.PRIORITY_HIGH
+import androidx.core.app.NotificationCompat.PRIORITY_LOW
 import androidx.core.content.ContextCompat.getColor
 import androidx.core.content.ContextCompat.getSystemService
+import org.onionshare.android.OnionShareApp
 import org.onionshare.android.R
 import javax.inject.Inject
 
-private const val CHANNEL_ID = "torSharingForegroundService"
-internal const val NOTIFICATION_ID = 1
+private const val CHANNEL_ID_FOREGROUND = "torSharingForegroundService"
+private const val CHANNEL_ID_SHARING = "torSharingStarted"
+internal const val NOTIFICATION_ID_FOREGROUND = 1
+internal const val NOTIFICATION_ID_SHARING = 2
 
 class OnionNotificationManager @Inject constructor(
     private val app: Application,
 ) {
 
     private val nm = getSystemService(app, NotificationManager::class.java)!!
+    private val pendingIntent: PendingIntent
+        get() = Intent(app, MainActivity::class.java).let { i ->
+            val pendingFlags = if (SDK_INT < 23) 0 else FLAG_IMMUTABLE
+            PendingIntent.getActivity(app, 0, i, pendingFlags)
+        }
+
+    private val isActive get() = (app as OnionShareApp).isActivityStarted
```

Comment:
This isn't a criticism or a request for you to change anything, but I'm still having trouble understanding when it would be idiomatic to declare a field with a getter for this kind of thing, rather than a private method that does the same thing as the getter. Is there some rule of thumb for this?

Reply:
I think the idiomatic Kotlin way is to declare a getter like I did above. If this "field" weren't private, this would actually get exposed to Java code as `getIsActive()` (or `isActive()`?).

I haven't read up on this, by my personal take is to use field getters when the getter is small and does not require any parameters.